scene/animation/animation_node_state_machine.cpp

```diff
@@ -628,14 +618,14 @@ double AnimationNodeStateMachinePlayback::process(AnimationNodeStateMachine *p_s
 }
 
 bool AnimationNodeStateMachinePlayback::_check_advance_condition(const Ref<AnimationNodeStateMachine> state_machine, const Ref<AnimationNodeStateMachineTransition> transition) const {
-	if (transition->has_auto_advance()) {
-		return true;
+	if (transition->get_advance_mode() != AnimationNodeStateMachineTransition::ADVANCE_MODE_AUTO) {
+		return false;
 	}
 
 	StringName advance_condition_name = transition->get_advance_condition_name();
 
-	if (advance_condition_name != StringName() && bool(state_machine->get_parameter(advance_condition_name))) {
-		return true;
+	if (advance_condition_name != StringName() && !bool(state_machine->get_parameter(advance_condition_name))) {
+		return false;
```

Comment:
Is this change in decision correct?

Reply:
I'm reading this over, and unless I've got something very wrong, it should be. The first check is to return false if the advance mode is not set to auto since only auto advance evaluates conditions and expressions (though this should probably instead be changed to ```transition->get_advance_mode() == AnimationNodeStateMachineTransition::ADVANCE_MODE_DISABLED)``` in case we want to add condition evaluation to the travel function in the future. As for the second check, this should also be correct in that first we're checking if the advance condition string is set to anything, then evaluating it and return false if the parameter is also false.